Assets/Scripts/Utilities/CCLogger.cs

```diff
@@ -41,20 +48,191 @@ private static string getSceneName()
+        eventDispatcher.LocationChanged.AddListener(LocationChanged);
+        eventDispatcher.AnnotationAdded.AddListener(AnnotationAdded);
+        eventDispatcher.AnnotationDeleted.AddListener(AnnotationDeleted);
+        eventDispatcher.AnnotationClear.AddListener(AnnotationClear);
+        eventDispatcher.PushPinSelected.AddListener(PushPinSelected);
+        eventDispatcher.PushPinUpdated.AddListener(PushPinUpdated);
+        eventDispatcher.DrawMode.AddListener(DrawMode);
+        eventDispatcher.SimulationTimeChanged.AddListener(SimulationTimeChanged);
+    }
+    private void LocationSelected(string newLocation)
+    {
+        _logAsync("UI Location Selected", newLocation);
+    }
+
+
+    private void LocationChanged(LatLng location, string place)
+    {
+        lastLocation = location;
+        string locationString = $"{location.ToDisplayString()} ({place})";
+        _logAsync("UI Location Changed", locationString);
+    }
+
+    // localPosition, localRotation, localScale, name 
+    private void AnnotationAdded(Vector3 position, Quaternion rotation, Vector3 scale, string name)
+    {
+        string msg = $"Name:{name}, P:{position.ToString()}, R:{rotation.ToString()}, S:{scale.ToString()}";
+        _logAsync("Annotation Added", msg);
+    }
+
+    private void AnnotationDeleted(string name)
+    {
+        _logAsync("Annotation Deleted", name);
+    }
+
+    private void AnnotationClear(string name)
+    {
+        _logAsync("Annotation Clear", name);
+    }
+
+    private void PushPinSelected(LatLng location, DateTime date)
+    {
+        lastLocation = location;
+        lastSimulationTime = date;
+        string msg = $"Location: {location.ToDisplayString()}, date:{date.ToString()}";
+        _logAsync("PushPin Selected", msg);
+    }
+
+    // Currentlocation, CurrentSimulationTime, cameraRotation
+    private void PushPinUpdated(LatLng location, DateTime date, Vector3 rotation)
+    {
+        lastLocation = location;
+        lastSimulationTime = date;
+        lastHeading = rotation;
+        string msg = $"Location:{location.ToDisplayString()}, date:{date.ToString()}, ";
+        _logAsync("PushPin Updated", msg);
+    }
+
+    private void DrawMode(bool isDrawing)
+    {
+        string msg = $"isDrawing:{isDrawing.ToString()}";
+        _logAsync("DrawMode Changed", msg);
+    }
+
+    public void SimulationTimeChanged() {
+        DateTime currentTime = SimulationManager.GetInstance().CurrentSimulationTime;
+        string msg = $"SimulationTime:{currentTime.ToString()}";
+        _logAsync("Simulation Time Changed", msg);
+    }
+
+
+
+    private static double getTimeMillies()
+    {
+        TimeSpan t = DateTime.UtcNow - new DateTime(1970, 1, 1);
+        return t.TotalMilliseconds;
+    }
+
+    private static string getSceneName()
+    {
+        UnityEngine.SceneManagement.Scene scene = UnityEngine.SceneManagement.SceneManager.GetActiveScene();
```

Comment:
can clean this up a bit if you add a Using statement up top to UnityEngine.SceneManagement then just return SceneManager.GetActiveScene().name;